### PR TITLE
Player Optimizations

### PR DESCRIPTION
**Pull request by alexduros**
This PR covers  : 

- [x] ~~putting `objectiron` step in worker (awaiting to do more).~~ The Web worker has been removed following the optimization of the `objectiron` script. It is so greatly optimized that posting data to the worker took more time than processing them. So it is no longer necessary.
- [x] optimizing manifest updates, so that it is possible to play usp 1.7.32 on web & chromecast
- [x] replaces `deep-equal` module with `fast-deep-equal` (as I couldn't find `fast-dashjs` 🙈 )
- [x] fix a dashjs bug which allow dash to switch on a profile with a non-supported codec
- [x] optimizing the `xml2json` script to greatly reduce the time required to parse XML manifest files.

### CODE REVIEW

File: src/dash/models/DashManifestModel.js [hunk 1]
```diff
@@ -195,15 +197,16 @@ function DashManifestModel() {
...
-        var adaptations = manifest.Period_asArray[periodIndex].AdaptationSet_asArray;
-        var i,
-            len;
-
-        for (i = 0, len = adaptations.length; i < len; i++) {
-            if (adaptations[i] === adaptation) {
-                return i;
+    function getIndexForAdaptation(realAdaptation, manifest, periodIndex) {
+        if (realAdaptation) {
+            const realAdaId = realAdaptation && realAdaptation.Representation && realAdaptation.Representation.id;
```
Comment (Gontran-Molotov) [inline src/dash/models/DashManifestModel.js]: No need to retest if `realAdaptation` is defined here.

Comment (alexduros) [reply src/dash/models/DashManifestModel.js]: 👍

File: src/streaming/ManifestLoader.js [hunk 2]
```diff
@@ -67,6 +72,13 @@ function ManifestLoader(config) {
             metricsModel: config.metricsModel,
             requestModifier: config.requestModifier
         });
+
+        try {
+            worker = new Worker('dash.worker.js');
+            worker.onmessage = onManifestParsed;
```
Comment (Gontran-Molotov) [inline src/streaming/ManifestLoader.js]: Shound also define `onError`?

File: src/dash/models/DashManifestModel.js [hunk 3]
```diff
@@ -248,8 +251,7 @@ function DashManifestModel() {
         return adaptations[0];
     }
 
-    function getCodec(adaptation) {
-        var representation = adaptation.Representation_asArray[0];
+    function getCodec(representation) {
```
Comment (happypoulp) [inline src/dash/models/DashManifestModel.js]: Need to fix tests for `getCodec` now that the signature has changed.

https://github.com/Dash-Industry-Forum/dash.js/blob/c1b6332ab95e15bd24d5c0729933773d8777f9f1/test/unit/dash.models.DashManifestModel.js#L217

**Comment by Gontran-Molotov** [comment]
Merged to the `molotov` branch in an other way.

**Pull request closed without merging**